tests/src/JIT/opt/FastTailCall/FastTailCallCandidates.cs

```diff
@@ -0,0 +1,1150 @@
+            return CalleeSimpleHFACase(one, 
+                                       two,
+                                       three,
+                                       four,
+                                       a,
+                                       b,
+                                       one,
+                                       two);
+        }
+        else
+        {
+            double b = one + 1599;
+            double a = one + 16;
+            return CalleeSimpleHFACase(two, 
+                                       one,
+                                       three,
+                                       four,
+                                       a,
+                                       b,
+                                       two,
+                                       one);
+        }
+    }
+
+    /// <summary>
+    /// Possible to fastTailCall only on arm64
+    /// </summary>
+    /// <remarks>
+    ///
+    /// This test case is really only interesting on arm64
+    ///
+    /// Arm64:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// x64 *nix:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// Arm64 can fastTailCall while x64 linux cannot
+    ///
+    /// Return 100 is a pass.
+    /// Return 108 is a failure.
+    ///
+    /// </remarks>
+    public static int CalleeHFAStackSpace(double one,
+                                          double two,
+                                          double three,
+                                          double four,
+                                          double five,
+                                          double six,
+                                          double seven,
+                                          double eight,
+                                          double nine,
+                                          double ten)
+    {
+        int count = 0;
+        for (double i = 0; i < one; ++i)
+        {
+            if (i % 2 == 0)
+            {
+                ++count;
+            }
+        }
+
+        if (count == 1)
+        {
+            return 100;
+        }
+
+        else
+        {
+            return 108;
+        }
+    }
+
+    /// <summary>
+    /// Possible to fastTailCall only on arm64
+    /// </summary>
+    /// <remarks>
+    ///
+    /// This test case is really only interesting on arm64
+    ///
+    /// Arm64:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// x64 *nix:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// Arm64 can fastTailCall while x64 linux cannot
```

Comment:
It seems like that is an implementation limitation, not a fundamental limitation, based on those comments. If so, this shouldn't say "cannot", but "won't, due to implementation limitation"

---
In reply to: [133308337](https://github.com/dotnet/coreclr/pull/12489#discussion_r133308337) [](ancestors = 133308337)